kibble/api/cache.go

```diff
@@ -151,11 +151,16 @@ func Get(cfg *models.Config, url string) ([]byte, error) {
 	}
 	defer resp.Body.Close()
 
+	body, err := ioutil.ReadAll(resp.Body)
+
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("request failed %s status code:%d", url, resp.StatusCode)
+		if err != nil {
```

Comment:
error from `ioutil.ReadAll`  and status code check are two different things. So it is better to check the error from `ioutil.ReadAll` separately and change the error message to something like `error reading response`.

Also, we are losing the error details from `ioutil.ReadAll`. Maybe log it or wrap it in the error we are returning?

Reply:
@josejibin Updated :)